pkg/external_repos/introspect_test.go

```diff
@@ -244,34 +304,61 @@ func TestUpdateIntrospectionStatusMetadata(t *testing.T) {
 				LastIntrospectionSuccessTime: &timestamp,
 				LastIntrospectionError:       pointy.String(""),
 				Status:                       config.StatusValid,
+				PackageCount:                 *pointy.Int(100),
 			},
 		},
 	}
 
-	for i := 0; i < len(testCases); i++ {
+	for _, testCase := range testCases {
+		t.Log(testCase.name)
+
 		var givenErr string
-		if testCases[i].given.err != nil {
-			givenErr = testCases[i].given.err.Error()
+		if testCase.given.err != nil {
+			givenErr = testCase.given.err.Error()
 		}
-		givenCount := testCases[i].given.count
-		givenStatus := testCases[i].given.status
-
-		expectedErr := testCases[i].expected.LastIntrospectionError
-		expectedStatus := testCases[i].expected.Status
-		expectedTime := testCases[i].expected.LastIntrospectionTime
-		expectedSuccessTime := testCases[i].expected.LastIntrospectionSuccessTime
-		expectedUpdateTime := testCases[i].expected.LastIntrospectionUpdateTime
 
 		repoIn := dao.Repository{
 			LastIntrospectionError: &givenErr,
-			Status:                 givenStatus,
+			Status:                 testCase.given.status,
+			PackageCount:           100,
+		}
+
+		result := updateIntrospectionStatusMetadata(
+			repoIn,
+			testCase.given.count,
+			testCase.given.err,
+			&timestamp)
+
+		if testCase.expected.LastIntrospectionError != nil {
+			require.NotNil(t, result.LastIntrospectionError)
+			assert.Equal(t, *testCase.expected.LastIntrospectionError, *result.LastIntrospectionError)
+		} else {
+			assert.Nil(t, result.LastIntrospectionError)
+		}
```

Comment:
What is the purpose of the if/else blocks here? Isn't it sufficient just to test that expected and result are equal? If expected is nil, result should be nil. If expected as some value, result should have that value.

Reply:
you are right; I was thinking about them that only the pointer were being compared; and in the case of leverage the data referenced, I was adding a require before. I have dig into the Equal, and it makes a deepequal of the fields using reflection, so when the pointers differs, it try to compare the data they are pointing out. That simplify many of the tests. Thanks mate!